libvirt/pool.go

```diff
@@ -0,0 +1,113 @@
+package libvirt
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	libvirt "github.com/libvirt/libvirt-go"
+)
+
+const (
+	poolExistsID    = "EXISTS"
+	poolNotExistsID = "NOT-EXISTS"
+)
+
+// poolExists returns "EXISTS" or "NOT-EXISTS" depending on the current pool existence
+func poolExists(virConn *libvirt.Connect, uuid string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		pool, err := virConn.LookupStoragePoolByUUIDString(uuid)
+		if err != nil {
+			if err.(libvirt.Error).Code == libvirt.ERR_NO_STORAGE_POOL {
+				log.Printf("Pool %s does not exist", uuid)
+				return virConn, "NOT-EXISTS", nil
+			}
+			log.Printf("Pool %s: error: %s", uuid, err.(libvirt.Error).Message)
+		}
+		if pool != nil {
+			defer pool.Free()
+		}
+		return virConn, poolExistsID, err
+	}
+}
+
+// poolWaitForExists waits for a storage pool to be up and timeout after 5 minutes.
+func poolWaitForExists(virConn *libvirt.Connect, uuid string) error {
+	log.Printf("Waiting for pool %s to be active...", uuid)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{poolNotExistsID},
+		Target:     []string{poolExistsID},
+		Refresh:    poolExists(virConn, uuid),
+		Timeout:    1 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for pool to reach EXISTS state: %s", err)
+	}
+	return nil
+}
+
+// poolWaitDeleted waits for a storage pool to be removed
+func poolWaitDeleted(virConn *libvirt.Connect, uuid string) error {
+	log.Printf("Waiting for pool %s to be deleted...", uuid)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{poolExistsID},
+		Target:     []string{poolNotExistsID},
+		Refresh:    poolExists(virConn, uuid),
+		Timeout:    1 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for pool to reach NOT-EXISTS state: %s", err)
```

Comment:
i was thinking if we could improve this Error message. 

This error is something that an User will read if something goes wrong and it seems more for a devel perspective.

I agree that we can keep it but imho we could improve the UX.

From experience we had in the past for the network IP the same `devel` message and at the end we had lot of issues open because the message was not user friendly enough, so an user could not take an action on it

`` Unexpected error during pool destroy operation. The pool was not deleted

Reply:
Sure, I'll change. Just FYI, this is copy&paste from the similar code for volume resource.

Reply:
ok thank you zeenix. I think we didn't spot it out. imho we can improve also the other errMsg. But not worth with PR imho. thank you for this :+1:

Reply:
I updated the messages and also sent the exact error to the log. PTALA.